// mpesa_express_api_service/dist/org.wso2.client.model.MpesaExpressAPIService/STKPushResponse.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _STKPushResponseHeader = _interopRequireDefault(require("./STKPushResponseHeader"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _typeof(obj) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (obj) { return typeof obj; } : function (obj) { return obj && "function" == typeof Symbol && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }, _typeof(obj); }
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, _toPropertyKey(descriptor.key), descriptor); } }
function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }
function _toPropertyKey(arg) { var key = _toPrimitive(arg, "string"); return _typeof(key) === "symbol" ? key : String(key); }
function _toPrimitive(input, hint) { if (_typeof(input) !== "object" || input === null) return input; var prim = input[Symbol.toPrimitive]; if (prim !== undefined) { var res = prim.call(input, hint || "default"); if (_typeof(res) !== "object") return res; throw new TypeError("@@toPrimitive must return a primitive value."); } return (hint === "string" ? String : Number)(input); } /**
                                                                                                                                                                                                                                                                                                                                                                                               * MpesaExpressAPIService
                                                                                                                                                                                                                                                                                                                                                                                               * Mpesa Express Checkout API
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * The version of the OpenAPI document: 1.0.0
                                                                                                                                                                                                                                                                                                                                                                                               * 
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
                                                                                                                                                                                                                                                                                                                                                                                               * https://openapi-generator.tech
                                                                                                                                                                                                                                                                                                                                                                                               * Do not edit the class manually.
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               */
/**
 * The STKPushResponse model module.
 * @module org.wso2.client.model.MpesaExpressAPIService/STKPushResponse
 * @version 1.0.0
 */
var STKPushResponse = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>STKPushResponse</code>.
   * @alias module:org.wso2.client.model.MpesaExpressAPIService/STKPushResponse
   * @param header {module:org.wso2.client.model.MpesaExpressAPIService/STKPushResponseHeader} 
   * @param response {Object} 
   */
  function STKPushResponse(header, response) {
    _classCallCheck(this, STKPushResponse);
    STKPushResponse.initialize(this, header, response);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  _createClass(STKPushResponse, null, [{
    key: "initialize",
    value: function initialize(obj, header, response) {
      obj['header'] = header;
      obj['response'] = response;
    }

    /**
     * Constructs a <code>STKPushResponse</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:org.wso2.client.model.MpesaExpressAPIService/STKPushResponse} obj Optional instance to populate.
     * @return {module:org.wso2.client.model.MpesaExpressAPIService/STKPushResponse} The populated <code>STKPushResponse</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new STKPushResponse();
        if (data.hasOwnProperty('header')) {
          obj['header'] = _STKPushResponseHeader["default"].constructFromObject(data['header']);
        }
        if (data.hasOwnProperty('response')) {
          obj['response'] = _ApiClient["default"].convertToType(data['response'], Object);
        }
      }
      return obj;
    }
  }]);
  return STKPushResponse;
}();
/**
 * @member {module:org.wso2.client.model.MpesaExpressAPIService/STKPushResponseHeader} header
 */
STKPushResponse.prototype['header'] = undefined;

/**
 * @member {Object} response
 */
STKPushResponse.prototype['response'] = undefined;
var _default = STKPushResponse;
exports["default"] = _default;
